internal/util/bitvect_test.go

```diff
@@ -53,80 +37,228 @@ func sampleRandomWide(n int) [][]byte {
 	return matrix
 }
 
-func TestUnReRavelingTall(t *testing.T) {
-	trange := []int{512, 512 * 2, 512 * 3, 512 * 4}
+// sampleZebraBlock creates a 512x512 bit block where every bit position
+// alternates between 0 and 1. When transposed, this block should
+// consists of rows of all 0s alternating with rows of all 1s.
+func sampleZebraBlock() BitVect {
+	zebraBlock2D := make([][]byte, 512)
 	var b BitVect
-	for _, a := range trange {
-		matrix := sampleRandomTall(a)
+	for row := range zebraBlock2D {
+		zebraBlock2D[row] = make([]byte, 64)
+		for c := 0; c < 64; c++ {
+			zebraBlock2D[row][c] = 0b01010101
+		}
+	}
+	b.unravelTall(zebraBlock2D, 0)
+	return b
+}
+
+// Property-based tests
+
+type tallMatrix struct {
+	matrix [][]byte
+}
+
+// Generate creates a struct containing two duplicate
+// copies of a pseudorandom tall matrix which has 512
+// columns and a multiple of 512 rows.
+func (tallMatrix) Generate(r *rand.Rand, size int) reflect.Value {
+	var tall tallMatrix
+	tall.matrix = sampleRandomTall(Pad(size, 512))
```

Comment:
Generate() takes an instance of Rand (is it seeded? does it need to be?) but doesn't pass it to sampleRandomTall(). Same seems to be the case for sampleRandomWide() as well.

Reply:
@xcapaldi You can create a `quick.Config` and give it a seeded `*rand.Rand`. This same `*rand.Rand` will be used for `Generate` downstream, if you pass the `quick.Config` to `quick.Check()`.

Reply:
Actually the default config is already seeded: https://cs.opensource.google/go/go/+/refs/tags/go1.17.5:src/testing/quick/quick.go;l=201

Reply:
actually, it's seeded automatically. @jvsg see https://cs.opensource.google/go/go/+/refs/tags/go1.17.5:src/testing/quick/quick.go;l=201

Reply:
However, you are right @jvsg that the randomness is not passed to `sampleRandomTall`. Also my understanding of the size hint was not correct so I'm fixing that now.

Reply:
Okay @juanli16 @jvsg I've updated the tests such that the matrices are truly random length and contain random elements. I realized this error also affected property-based testing in the bit utilities so I've updated it there as well.